src/graph_gui.cpp

```diff
@@ -358,25 +359,25 @@ struct BaseGraphWindow : Window {
 			if (TimerGameEconomy::UsingWallclockUnits()) {
 				grid_colour = (i % 4 == 0) ? GRAPH_YEAR_LINE_COLOUR : GRAPH_GRID_COLOUR;
 			}
-			GfxFillRect(x, r.top, x, r.bottom, grid_colour);
+			GfxFillRect(x, r.top, x + gridline_width - 1, r.bottom, grid_colour);
 			x += x_sep;
 		}
 
 		/* Draw the horizontal grid lines. */
 		y = r.bottom;
 
 		for (int i = 0; i < (num_hori_lines + 1); i++) {
-			GfxFillRect(r.left - ScaleGUITrad(3), y, r.left - 1, y, GRAPH_AXIS_LINE_COLOUR);
-			GfxFillRect(r.left, y, r.right, y, GRAPH_GRID_COLOUR);
+			GfxFillRect(r.left - ScaleGUITrad(3), y, r.left - 1, y + gridline_width - 1, GRAPH_AXIS_LINE_COLOUR);
+			GfxFillRect(r.left, y, r.right + gridline_width - 1, y + gridline_width - 1, GRAPH_GRID_COLOUR);
```

Comment:
If you are going to `gridline_width - 1` everything, why not just do it once at the start, and not have to do it everywhere else? :)

Reply:
I'd argue it's better for readability, as the drawn line width is `gridline_width` pixels wide. The `- 1` is only needed because of `GfxFillRect` behaviour.

But, not a strong opinion!

Reply:
The longer I look at the original code, the more weirded out I am. It feels very odd, in the way it is structured. And that makes me even more confused about the "sometimes - 1 but not always" of the old code. But that is not your problem, so I am not going to bother you with it :)

I am going to believe you on your word, that this is easier to understand, and let's leave it as it is :)

Reply:
In my experience this type of code always gets a bit messy. It's the combination of line lengths becoming dependent on line width, and the behaviour of `GfxFillRect` meaning when `left == right` the width is one rather than zero.